Documentation example does not work

The linked [documentation example](https://pkg.go.dev/go.uber.org/yarpc?tab=doc#example-Dispatcher.Register-JSON) does not work. Here's a minimal example:

```
package main

import (
        "context"
        "fmt"
        "log"

        "go.uber.org/yarpc"
        "go.uber.org/yarpc/encoding/json"
)

func main() {
        dispatcher := yarpc.NewDispatcher(yarpc.Config{
                Name: "yarpc-dev",
        })

        if err := dispatcher.Start(); err != nil {
                log.Fatal(err)
        }
        defer dispatcher.Stop()

        handler := func(ctx context.Context, key string) (string, error) {
                fmt.Println("key", key)

                return "value", nil
        }

        dispatcher.Register(json.Procedure("get", handler))
}
```

The error message:

```
$ go run main.go
panic: the second argument of the handler for "get" must be a struct pointer, a map[string]interface{}, or interface{}, and not: string

goroutine 1 [running]:
go.uber.org/yarpc/encoding/json.verifyInputSignature(0x96cc3f, 0x3, 0xa3b360, 0x8ccfc0, 0xc0002efc88, 0x7282b4)
        /home/huyage/go/pkg/mod/go.uber.org/yarpc@v1.44.0/encoding/json/register.go:197 +0x2a0
go.uber.org/yarpc/encoding/json.verifyUnarySignature(0x96cc3f, 0x3, 0xa3b360, 0x8ccfc0, 0x946360, 0x0)
        /home/huyage/go/pkg/mod/go.uber.org/yarpc@v1.44.0/encoding/json/register.go:117 +0x67
go.uber.org/yarpc/encoding/json.wrapUnaryHandler(0x96cc3f, 0x3, 0x8ccfc0, 0x990400, 0x0, 0x0)
        /home/huyage/go/pkg/mod/go.uber.org/yarpc@v1.44.0/encoding/json/register.go:86 +0x86
go.uber.org/yarpc/encoding/json.Procedure(...)
        /home/huyage/go/pkg/mod/go.uber.org/yarpc@v1.44.0/encoding/json/register.go:59
main.main()
        /home/huyage/dev/yarpcdev/main.go:28 +0x138
exit status 2
```

--- a/dispatcher_example_test.go
+++ b/dispatcher_example_test.go
@@ -31,11 +31,12 @@ import (
 )
 
 func ExampleDispatcher_minimal() {
-	dispatcher := yarpc.NewDispatcher(yarpc.Config{Name: "myFancyService"})
+	dispatcher := yarpc.NewDispatcher(yarpc.Config{Name: "my-fancy-service"})
 	if err := dispatcher.Start(); err != nil {
 		log.Fatal(err)
 	}
 	defer dispatcher.Stop()
+	// Output:
 }
 
 // global dispatcher used in the registration examples
@@ -47,16 +48,26 @@ func ExampleDispatcher_Register_raw() {
 	}
 
 	dispatcher.Register(raw.Procedure("echo", handler))
+	// Output:
+}
+
+type Request struct {
+	key string
+}
+
+type Response struct {
+	val string
 }
 
 // Excuse the weird naming of this function. This lets is show as "JSON"
 // rather than "Json"
 
 func ExampleDispatcher_Register_jSON() {
-	handler := func(ctx context.Context, key string) (string, error) {
-		fmt.Println("key", key)
-		return "value", nil
+	handler := func(ctx context.Context, req *Request) (*Response, error) {
+		fmt.Println("key", req.key)
+		return &Response{val: "value"}, nil
 	}
 
 	dispatcher.Register(json.Procedure("get", handler))
+	// Output:
 }

